### PR TITLE
packaging: simplify spec file for RHEL/Fedora

### PR DESCRIPTION
**Pull request by czanik**
Simplify spec file by removing obsolete technologies:
- remove RHEL 6 support
- remove Python 2 support
- keep Java support, but remove Java-based drivers (HDFS,
  Elasticsearch, etc.)
- minor reorganization to make it more readable

Signed-off-by: Peter Czanik <peter@czanik.hu>

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by MrAnno** [comment]
@kira-syslogng retest this please

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by Kokan** [comment]
@kira-syslogng retest this please

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: packaging/rhel/syslog-ng.spec [hunk 1]
```diff
@@ -92,7 +58,6 @@ BuildRequires: pcre-devel
 BuildRequires: libuuid-devel
 BuildRequires: libesmtp-devel
 BuildRequires: libcurl-devel
-BuildRequires: tzdata
```
[the remark this answers is not in the document]
Comment (czanik) [reply packaging/rhel/syslog-ng.spec:95]: I just double checked, it was not part of my syslog-ng spec file for at least the past two years. And it is installed automatically even without specifying it in the spec file.

[the remark this answers is not in the document]
Comment (Kokan) [reply packaging/rhel/syslog-ng.spec:95]: Discussed with @czanik, this is always provided, and those usually are omitted.

**Review by Kokan — APPROVED** [review]

**Comment by bazsi** [comment]
The tzdata package was iirc missing in our dbld images.

**Comment by bazsi** [comment]
@kira-syslogng build packages please;

**Comment by kira-syslogng** [comment]
Package Build SUCCESS

Packages are available at https://ci.syslog-ng.com/jenkins/job/kira-dbld-pull-requests/735/artifact/

**Review by lbudai — APPROVED** [review]

**Pull request merged**